customer-commons/src/test/java/no/unit/nva/customer/service/impl/DynamoDBCustomerServiceTest.java

```diff
@@ -0,0 +1,207 @@
+import no.unit.nva.customer.model.Customer;
+import no.unit.nva.customer.service.CustomerService;
+import nva.commons.utils.Environment;
+import org.junit.Rule;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.migrationsupport.rules.EnableRuleMigrationSupport;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@EnableRuleMigrationSupport
+public class DynamoDBCustomerServiceTest {
+
+    public static final String NVA_CUSTOMERS_TEST = "nva_customers_test";
+    @Rule
+    public CustomerDynamoDBLocal db =  new CustomerDynamoDBLocal();
+
+    private ObjectMapper objectMapper = new ObjectMapperConfig().objectMapper;
+    private DynamoDBCustomerService service;
+    private Environment environment;
+    private AmazonDynamoDB client;
+
+    /**
+     * Set up environment.
+     */
+    @BeforeEach
+    public void setUp() {
+        client = DynamoDBEmbedded.create().amazonDynamoDB();
+        environment = mock(Environment.class);
+        service = new DynamoDBCustomerService(
+                objectMapper,
+                db.getTable()
+        );
+    }
+
+    @Test
+    public void testConstructor() {
+        when(environment.readEnv(DynamoDBCustomerService.TABLE_NAME)).thenReturn(NVA_CUSTOMERS_TEST);
+        CustomerService serviceWithTableNameFromEnv = new DynamoDBCustomerService(client, objectMapper, environment);;
+        assertNotNull(serviceWithTableNameFromEnv);
+    }
+
+    @Test
+    public void createNewCustomer() throws Exception {
+        Customer customer = getNewCustomer();
+        Customer createdCustomer = service.createCustomer(customer);
+
+        Assertions.assertNotNull(createdCustomer.getIdentifier());
+        Assertions.assertEquals(customer, createdCustomer);
+    }
+
+    @Test
+    public void updateExistingCustomer() throws Exception {
+        String newName = "New name";
+        Customer customer = getNewCustomer();
+        Customer createdCustomer = service.createCustomer(customer);
+        Assertions.assertNotEquals(newName, createdCustomer.getName());
+
+        createdCustomer.setName(newName);
+        Customer updatedCustomer = service.updateCustomer(createdCustomer.getIdentifier(), createdCustomer);
+        Assertions.assertEquals(newName, updatedCustomer.getName());
+        Assertions.assertNotEquals(customer.getModifiedDate(), updatedCustomer.getModifiedDate());
+        Assertions.assertEquals(customer.getCreatedDate(), updatedCustomer.getCreatedDate());
+
+    }
+
+    @Test
+    public void getExistingCustomer() throws Exception {
```

Comment:
what happens when there is no customer? shouldn't we test that? Also test name should be improved

Reply:
There is a test for that `getCustomerNotFound`